### PR TITLE
[unrar] Add vcpkg-cmake-wrapper and usage

### LINKED ISSUE #22688: [unrar] Add unrarConfig.cmake
**Is your feature request related to a problem? Please describe.**
Although `unrar` builds fine, it doesn't have an `unrarConfig.cmake`, which leads to CMake not finding the library using `find_library`.

**Proposed solution**
Add an `unrarConfig.cmake`

### PR DESCRIPTION
**Pull request by LilyWangLL**
Fixes #22688
Add `vcpkg-cmake-wrapper.cmake` and `usage` for `unrar`.

### CODE REVIEW

File: ports/unrar/usage [hunk 1]
```diff
@@ -0,0 +1,5 @@
+The package unrar provides CMake integration:
+
+    find_package(unrar REQUIRED)
+    target_include_directories(main PRIVATE ${UNRAR_INCLUDE_DIR})
+    target_link_libraries(main PRIVATE ${UNRAR_LIBRARIES})
```
Comment (dg0yt) [inline ports/unrar/usage:5]: Missing line break at the end.

**Review by dg0yt — CHANGES_REQUESTED** [review]
I see the good intentions but in this form, it is invalid.

- There is no official (CMake) Find module for (un)rar, so there can't be a wrapper.
- If desired, you can install a CMake config file. But it must be in the `unofficial` namespace.

File: ports/unrar/portfile.cmake [hunk 2]
```diff
@@ -36,5 +36,8 @@ file(INSTALL "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}-dbg/unrar.lib"  DESTIN
 
 vcpkg_copy_pdbs()
 
+file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
```
Comment (ras0219-msft) [inline ports/unrar/portfile.cmake:39]: There's no need to use the heavy hammer of `vcpkg-cmake-wrapper.cmake` here. Instead, this should create a `${CURRENT_PACKAGES_DIR}/share/unrar/unrar-config.cmake`.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]

File: ports/unrar/Config.cmake.in [hunk 3]
```diff
@@ -0,0 +1,21 @@
+
+get_filename_component(_unrar_root "${CMAKE_CURRENT_LIST_FILE}" PATH)
+get_filename_component(_unrar_root "${_unrar_root}" PATH)
+get_filename_component(_unrar_root "${_unrar_root}" PATH)
+
+set(_unrar_lib "${_unrar_root}/lib/unrar.lib")
```
Comment (LilyWangLL) [inline ports/unrar/Config.cmake.in:6]: Due to `unrar` use build `vcxproj` by msbuild, it only support target `windows`, so set `_unrar_lib` is `unrar.lib`.

File: ports/unrar/Config.cmake.in [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    add_library(unrar INTERFACE IMPORTED)
+    set_target_properties(unrar PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${_unrar_root}/include")
+    set_target_properties(unrar PROPERTIES IMPORTED_LOCATION "${_unrar_lib}")
+    set_property(TARGET unrar APPEND PROPERTY IMPORTED_CONFIGURATIONS)
+
+else()
+
+    set(unrar_FOUND FALSE)
+
```
Comment (JackBoosY) [inline ports/unrar/Config.cmake.in:17]:
```suggestion
    set(unrar_FOUND TRUE)
else()
    set(unrar_FOUND FALSE)
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Review by dg0yt — CHANGES_REQUESTED** [review]
While it is not a cmake project, I stll think the config should be in the `unofficial` namespace. Using it creates adependency on building with vcpkg.

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**